pkg/nsx/client.go

```diff
@@ -58,30 +62,64 @@ func GetClient(cf *config.NSXOperatorConfig) *Client {
 	vspherelog.SetLogger(logger)
 	c := NewConfig(strings.Join(cf.NsxApiManagers, ","), cf.NsxApiUser, cf.NsxApiPassword, "", 10, 3, 20, 20, true, true, true, ratelimiter.AIMD, cf.GetTokenProvider(), nil, cf.Thumbprint)
 	cluster, _ := NewCluster(c)
-	nsxVersion, err := cluster.GetVersion()
-	if err != nil {
-		log.Error(err, "get version error")
-		return nil
-	}
-	err = nsxVersion.Validate(minVersion)
-	if err != nil {
-		log.Error(err, "validate version error")
-		return nil
-	}
+
 	queryClient := search.NewQueryClient(restConnector(cluster))
 	groupClient := domains.NewGroupsClient(restConnector(cluster))
 	securityClient := domains.NewSecurityPoliciesClient(restConnector(cluster))
 	ruleClient := security_policies.NewRulesClient(restConnector(cluster))
 	nsxChecker := &NSXHealthChecker{
 		cluster: cluster,
 	}
-	return &Client{
+	nsxVersionChecker := &NSXVersionChecker{
+		cluster:          cluster,
+		versionSupported: false,
+	}
+
+	if nsxVersion.featureSupported(minVersion) {
+		nsxVersionChecker.versionSupported = true
+	} else {
+		err := errors.New("NSX version check failed")
+		log.Error(err, "SecurityPolicy feature is not supported", "current version", nsxVersion.NodeVersion, "required version", minVersion)
+	}
```

Comment:
Why it print the similar log with line 96-99?

Reply:
They are redundant code and are removed. Thanks for reminder.